File: app/src/main/java/bj/rxjavaexperimentation/model/release/Label.java

package bj.rxjavaexperimentation.model.release;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Label
{

    @SerializedName("catno")
    @Expose
    private String catno;
    @SerializedName("entity_type")
    @Expose
    private String entityType;
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("resource_url")
    @Expose
    private String resourceUrl;
    private String thumb;

    public String getCatno()
    {
        return catno;
    }

    public void setCatno(String catno)
    {
        this.catno = catno;
    }

    public String getEntityType()
    {
        return entityType;
    }

    public void setEntityType(String entityType)
    {
        this.entityType = entityType;
    }

    public String getId()
    {
        return id;
    }

    public void setId(String id)
    {
        this.id = id;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getResourceUrl()
    {
        return resourceUrl;
    }

    public void setResourceUrl(String resourceUrl)
    {
        this.resourceUrl = resourceUrl;
    }

    public void setThumb(String thumb)
    {
        this.thumb = thumb;
    }

    public String getThumb()
    {
        return thumb;
    }
}

File: app/src/main/java/bj/vinylbrowser/model/common/Label.kt
package bj.vinylbrowser.model.common

import android.os.Parcel
import android.os.Parcelable
import bj.vinylbrowser.model.release.Image
import com.google.gson.annotations.SerializedName

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class Label(@SerializedName("resource_url") var resourceUrl: String = "",
                 @SerializedName("entity_type") var entityType: String = "",
                 var catno: String = "",
                 var id: String = "",
                 var name: String = "",
                 var profile: String = "",
                 @SerializedName("releases_url") var releaseUrl: String = "",
                 var uri: String = "",
                 var images: List<Image> = emptyList(),
                 @SerializedName("data_quality") var dataQuality: String = "",
                 var thumb: String = "") : Parcelable {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(resourceUrl)
        dest.writeString(entityType)
        dest.writeString(catno)
        dest.writeString(id)
        dest.writeString(name)
        dest.writeString(profile)
        dest.writeString(releaseUrl)
        dest.writeString(uri)
        dest.writeList(images)
        dest.writeString(dataQuality)
        dest.writeString(thumb)
    }

    override fun describeContents(): Int {
        return 0
    }
}